view_student.js

```diff
@@ -552,18 +552,35 @@ function realtimequiz_process_contents(httpRequest) {
                     }
                 }
             }
-        } else {
-            // Server responded with anything other than OK
-            // Decided just to silently resend the request - if the connection dies altoghether, the user can navigate
-            // to another page to stop the requests
-
-            //alert(realtimequiz.text['httperror']+httpRequest.status);
-            //if (confirm(realtimequiz.text['tryagain'])) {
-            realtimequiz_delayed_request("realtimequiz_resend_request()", 700);
-            //} else {
-            //realtimequiz_return_course();
-            // }
+            return;
+        } else if (httpRequest.status == 403 || httpRequest.status == 500) {
+            var jsonresp;
+            try {
+                jsonresp = JSON.parse(httpRequest.responseText);
+            } catch {
```

Comment:
For reference, this should be:
```
} catch (e) {
```
Otherwise it breaks the javascript. Thankfully Travis flagged this up when the behat tests broke.

Reply:
I'm guessing some browsers are more forgiving than others on that front - my chrome instance didn't emit any warning during testing. Thanks for the fyi :)

Reply:
Using chrome here (67 / Linux) and it was really not happy, neither was Firefox. Both of them failed to show the teacher options or the questions, there was just an empty box where they should have been. Maybe you didn't clear the browser caches and were still running the old code? Either way, it is fixed now.